platform/linux-generic/odp_ishm.c

```diff
@@ -909,22 +910,38 @@ static int do_unmap(void *start, uint64_t size, uint32_t flags,
  * Returns the index of the found block (if any) or -1 if none.
  * Mutex must be assured by the caller.
  */
-static int find_block_by_name(const char *name)
+static int find_block_by_name(const char *name, odp_bool_t use_param_name)
 {
 	int i;
 
 	if (name == NULL || name[0] == 0)
 		return -1;
 
 	for (i = 0; i < ISHM_MAX_NB_BLOCKS; i++) {
-		if ((ishm_tbl->block[i].len) &&
-		    (strcmp(name, ishm_tbl->block[i].name) == 0))
+		const char *block_name = use_param_name ?
+					 ishm_tbl->block[i].param_name :
+					 ishm_tbl->block[i].name;
+
+		if (ishm_tbl->block[i].len && strcmp(name, block_name) == 0)
 			return i;
 	}
 
 	return -1;
 }
 
+/*
+ * Make application given memory block name unique
+ */
+static void make_name_unique(char *block_name, const char *name, size_t sz)
+{
+	int i = 0;
+
+	_odp_strcpy(block_name, name, sz);
+
+	while (find_block_by_name(block_name, false) >= 0)
+		_odp_snprint(block_name, sz, "%d-%s", i++, name);
```

Comment:
Now that I noticed that the create_seq_string() and ishm_tbl->dev_seq are already there,  maybe (?) this could just generate the unique name using dev_seq or similar and avoid the search?

Reply:
Changed in V3.